pkg/aws/policies.go

```diff
@@ -853,14 +854,23 @@ func (c *awsClient) ListOperatorRoles(version string) (map[string][]Role, error)
 					}
 					isTagged = true
 					operatorRole.Version = tagValue
+				
+				case tags.ClusterID:
+					tagValue := aws.StringValue(tag.Value)
+						if clusterId != "" && tagValue != clusterId {
+							skip = true
+							break
+						}
+						isTagged = true
+						operatorRole.ClusterID = tagValue 
 				}
-			}
 			if isTagged && !skip {
 				operatorRole.RoleName = aws.StringValue(role.RoleName)
 				operatorRole.RoleARN = aws.StringValue(role.Arn)
 				operatorMap[foundPrefix] = append(operatorMap[foundPrefix], operatorRole)
 			}
 		}
+		}
```

Comment:
By doing this you moved `if isTagged && !skip {` this condition to within the loop, that's why it's got duplicates in the map now